engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -157,10 +156,15 @@ public void onTerminate(
     }
 
     // test that completion condition can be evaluated correctly
-    final Either<Failure, Boolean> canBeCompletedOrFailure =
-        canBeCompletedWithCompletionCondition(element, flowScopeContext);
-    if (canBeCompletedOrFailure.isLeft()) {
-      return canBeCompletedOrFailure;
+    final Either<Failure, Boolean> satisfiedOrFailure =
```

Comment:
🔧 I think this variable name could be improved by mentioning the completion condition. That would make it more clear what it satisfies. Perhaps `satisfiesCompletionConditionOrFailure`